platform/ios/src/MGLMapView.mm

```diff
@@ -2358,6 +2360,81 @@ - (void)setShowsScale:(BOOL)showsScale
     }
 }
 
+- (void)setShowsBuildings:(BOOL)showsBuildings
+{
+    if  (_showsBuildings == showsBuildings) return;
+    
+    NSString *layerIdentifier = @"extrudedBuildings";
+    if (showsBuildings) {
+        MGLSource *source = [self.style sourceWithIdentifier:@"composite"];
+        if (source) {
+            
+            MGLFillExtrusionStyleLayer* layer = [[MGLFillExtrusionStyleLayer alloc] initWithIdentifier:layerIdentifier source:source];
+            layer.sourceLayerIdentifier = @"building";
```

Comment:
I'm not a fan of using these arbitrary strings here - it's not clear where they come from or what they should even be. Can we generate constants from the style schemas, and commit them?

Reply:
@julianrex `composite` and `building` comes from our style spec. 
I agree, it may be good to fetch from constants instead of strings but I'm not sure what other considerations are in place. @1ec5 @friedbunny do you have any recommendations?

Reply:
Source layer identifiers are defined by the source in TileJSON and documented [in HTML](https://www.mapbox.com/vector-tiles/mapbox-streets-v7/). We hard-code strings like this in the VoiceOver code too. There could be opportunities to make this logic configurable using plists, but I don’t think there’s a good way to automatically derive it from the sources.